### DISCUSSION TITLE
Execution Clients - Erigon and Reth

### DISCUSSION THREAD

**Discussion by andreclaro** [celo-org/celo-blockchain#2312] [Help (Q&A)] [answered]
Hello,

Which execution clients alternatives work with Celo?

Does Erigon and/or Reth work?

repos:
- https://github.com/ledgerwatch/erigon
- https://github.com/paradigmxyz/reth

Thanks!

**Comment by palango** [comment, marked as answer]
Hi @andreclaro, 
Celo implements things like a custom PoS consensus algorithm and payment of transactions fee with fee currencies.
Those features haven't been implemented on other clients yet, so Erigon and Reth don't work with the Celo chain right now.

**Reply by andreclaro** [reply]
Thanks @palango

**Reply by andreclaro** [reply]
@palango, Does snap sync (`--syncmode=snap`) work in Celo?
